numpy/core/tests/test_datetime.py

```diff
@@ -137,6 +138,23 @@ def test_datetime_casting_rules(self):
         assert_(not np.can_cast('M8[h]', 'M8', casting='same_kind'))
         assert_(not np.can_cast('M8[h]', 'M8', casting='safe'))
 
+        # regression tests related to gh-19631;
+        # test metric prefixes from seconds down to
+        # attoseconds for 1000x conversions
+        assert np.can_cast('M8[7000ms]', 'M8[7s]', casting='safe')
+        assert np.can_cast('M8[2000us]', 'M8[2ms]', casting='safe')
+        assert np.can_cast('M8[1000ns]', 'M8[us]', casting='safe')
+        assert np.can_cast('M8[5000ns]', 'M8[5us]', casting='safe')
+        assert np.can_cast('M8[2000ps]', 'M8[2ns]', casting='safe')
+        assert np.can_cast('M8[9000fs]', 'M8[9ps]', casting='safe')
+        assert np.can_cast('M8[1000as]', 'M8[1fs]', casting='safe')
+        # 10^6 conversions
+        assert np.can_cast('M8[2000000ps]', 'M8[2us]', casting='safe')
+        assert np.can_cast('M8[1000000as]', 'M8[1ps]', casting='safe')
+        # 10^9 conversions below 32-bit overflow limit
+        assert np.can_cast('M8[2000000000ps]', 'M8[2ms]', casting='safe')
+        assert np.can_cast('M8[1000000000as]', 'M8[1ns]', casting='safe')
```

Comment:
Could you start a new function for this and test both directions?  These only test smaller unit to larger unit.  Or is that the only thing we fix?

Reply:
Sure, I drafted in a test that does bidirectional casting checks instead of just small units to large units. Let me know what you think. Luckily, the other direction "just worked" for the cases I already had there.